src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -7945,48 +7945,27 @@ void SizeVRF(EnergyPlusData &state, int const VRFTUNum)
-            } else if (CapSizingMethod == FractionOfAutosizedCoolingCapacity) {
-                state.dataSize->DataFracOfAutosizedCoolingCapacity = state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledCoolingCapacity;
-                state.dataSize->DataScalableCapSizingON = true;
-            }
-        }
+        initCapSizingVars(state,
+                          CoolingCapacitySizing,
+                          state.dataSize->ZoneHVACSizing(zoneHVACIndex).CoolingCapMethod,
+                          EqSizing.SizingMethod(SizingMethod),
+                          state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledCoolingCapacity,
+                          EqSizing.CoolingCapacity,
+                          EqSizing.DesCoolingLoad,
+                          state.dataSize->DataScalableCapSizingON,
+                          state.dataSize->DataFracOfAutosizedCoolingCapacity);
 
         // initialize capacity sizing variables: heating
-        SizingMethod = HeatingCapacitySizing;
-        CapSizingMethod = state.dataSize->ZoneHVACSizing(zoneHVACIndex).HeatingCapMethod;
-        EqSizing.SizingMethod(SizingMethod) = CapSizingMethod;
-        if (CapSizingMethod == HeatingDesignCapacity || CapSizingMethod == CapacityPerFloorArea ||
-            CapSizingMethod == FractionOfAutosizedHeatingCapacity) {
-            if (CapSizingMethod == HeatingDesignCapacity) {
-                if (state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledHeatingCapacity > 0.0) {
-                    EqSizing.HeatingCapacity = true;
-                    EqSizing.DesHeatingLoad = state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledHeatingCapacity;
-                }
-            } else if (CapSizingMethod == CapacityPerFloorArea) {
-                EqSizing.HeatingCapacity = true;
-                EqSizing.DesHeatingLoad = state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledHeatingCapacity *
-                                          state.dataHeatBal->Zone(state.dataSize->DataZoneNumber).FloorArea;
-                state.dataSize->DataScalableCapSizingON = true;
-            } else if (CapSizingMethod == FractionOfAutosizedHeatingCapacity) {
-                state.dataSize->DataFracOfAutosizedHeatingCapacity = state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledHeatingCapacity;
-                state.dataSize->DataScalableCapSizingON = true;
-            }
-        }
+        initCapSizingVars(state,
+                          HeatingCapacitySizing,
+                          state.dataSize->ZoneHVACSizing(zoneHVACIndex).HeatingCapMethod,
+                          EqSizing.SizingMethod(SizingMethod),
+                          state.dataSize->ZoneHVACSizing(zoneHVACIndex).ScaledHeatingCapacity,
+                          EqSizing.HeatingCapacity,
+                          EqSizing.DesHeatingLoad,
+                          state.dataSize->DataScalableCapSizingON,
+                          state.dataSize->DataFracOfAutosizedHeatingCapacity);
+
```

Comment:
This will certainly avoid the cut-n-paste errors that caused this issue. Is there a chance that the other 12 parent objects will get this change (I searched for FractionOfAutosizedCoolingCapacity)? Or at the very least a new issue opened?

The remaining discussion is for future use.

Now to think outside the box even further... What is really needed is for this "scaling" code to be moved to the sizer class so it's in 1 place. So that each parent does not need this overhead. All the sizer code would need to know is state.dataHVACVarRefFlow->VRFTU(VRFTUNum).HVACSizingIndex. That is, of course, for another day. That said, maybe the arguments could be shortened to pass only the HVACSizingIndex? and a key?

    initSizingVars(state, sizingMethod, HVACSizingIndex) 

where sizingMethod = CoolingAirFlowSizing, HeatingAirFlowSizing, CoolingCapacitySizing, or HeatingCapacitySizing (did I miss anything?). Maybe that's too much to ask for this issue but it's good to be thinking in that direction (for the sake of moving this to the sizing routines). In this future code I still want to be able to call the sizing like this (maybe with an HVACSizingIndex argument?) 

    sizingCoolingAirFlow.size(state, TempSize, errorsFound)

but let the sizer handle the scaling. If I were calling CoolingAirFlowSizing I could look up the "cooling air flow" methods on the HVACSizing object because I already know I am sizing cooling air flow. Or set up all scalable sizing in the base class and each sizer uses what is relevant. This was considered when the sizer method was first developed but I decided to punt. Here's a hook, and then see the bottom of initializeWithinEP where this was initially started and then abandoned:

    void BaseSizerWithScalableInputs::setHVACSizingIndexData(int const index)
    {
        this->zoneHVACSizingIndex = index;
    }

Reply:
@rraustad I like your line of thinking here.  It would be *really* nice if this could be standardized across the various models that do this and your thoughts and expertise here are clearly very valuable.

I took a quick look at the various other occurrences in the code using a search for FractionOfAutosizedCoolingCapacity like you did.  It seems like there are maybe a couple of cases where the code at first glance at least looks fairly similar to what existed here before the fix was implemented.  However, there are other cases where additional steps are being taken.  The question is then: how do we accommodate that and still get something that avoids the duplication of code across numerous instances.  I really would like to see that happen, but I'm also not sure how that would work at this time.  I wonder if that is why you "punted" as you said in your comment?

From my perspective, I am glad that my "value add" to the code here got this conversation started (or restarted?) and might yield other improvements to the code.  It would have been easy to just change the one line and move on, but I thought it made more sense to also come up with something that avoids the problem.  However, given my relatively low level of experience with the VRF and sizing code, I'm not sure that I'd be the most efficient person to handle anything beyond this.  So, I think my vote would be to not attempt to bump the complexity of this up and do more changes here but to definitely open up a new issue that includes your comments here.

Thoughts?  Any of the other flagged reviewers want to offer their opinions here?

Reply:
@RKStrand I'm fine with leaving this as-is. I punted because the sizing functions were enough work on their own without adding in the rewriting and associated testing of scalable sizing in the zone equipment. I did leave the hook in as a starting place for this "sizable" refactor.